lib/rules/no-constant-condition.js

```diff
@@ -32,11 +32,15 @@ module.exports = function(context) {
                 return true;
 
             case "UnaryExpression":
-                return isConstant(node.argument);
+                return node.operator === "typeof" || isConstant(node.argument);
 
             case "BinaryExpression":
             case "LogicalExpression":
-                return isConstant(node.left) && isConstant(node.right) && node.operator !== "in";
+                return node.left.operator !== "typeof" &&
```

Comment:
I am already checking node.operator for typeof value and returning true. Probably need to pass extra param . I will recheck the logic if something can be done to prevent deep inspection .